ext/bg/js/translator.js

```diff
@@ -45,9 +46,113 @@ class Translator {
             await this.buildTermFrequencies(definition, titles);
         }
 
+        if (options.general.compactTags) {
+            for (const definition of definitionsGrouped) {
+                dictTermsCompressTags(definition.definitions);
+            }
+        }
+
         return {length, definitions: definitionsGrouped};
     }
 
+    async findTermsMerged(text, dictionaries, alphanumeric) {
+        const options = await apiOptionsGet();
+        const mainDictionary = Object.keys(options.dictionaries).filter(dict => options.dictionaries[dict].main).concat([''])[0];
+        const secondarySearchTitles = Object.keys(options.dictionaries).filter(dict => options.dictionaries[dict].allowSecondarySearches);
+        const titles = Object.keys(dictionaries);
+        const {length, definitions} = await this.findTerms(text, dictionaries, alphanumeric);
+
+        const definitionsBySequence = dictTermsMergeBySequence(definitions, mainDictionary);
+
+        const definitionsMerged = [];
+        const mergedByTermIndices = new Set();
+        for (const sequence in definitionsBySequence) {
+            if (!(sequence >= 0)) {
+                continue;
+            }
+
+            const result = definitionsBySequence[sequence];
+
+            const rawDefinitionsBySequence = await this.database.findTermsBySequence(Number(sequence), mainDictionary);
+            const definitionsByGloss = dictTermsMergeByGloss(result, rawDefinitionsBySequence);
+
+            const secondarySearchResults = [];
+            if (secondarySearchTitles.length) {
```

Comment:
Minor: prefer explicit boolean expressions for things that aren't checking whether or not something is defined, meaning this should check `secondarySearchTtiles.length > 0`. In the coding style of Yomichan it makes it look like you are testing if `length` is `undefined` or `null`.